### PR TITLE
Fix(eos_cli_config_gen): AAA authentication on console fix

### LINKED ISSUE #1099: AAA authentication on console miss-configured
## Issue Type

- [x] Bug Report
- [x] EOS Configuration rendering error

## Summary

Seems a mistake was made when implementing AAA authentication commands with console, probably related to FN39.
Under mentioned FN, it is informed that "aaa authorization console" command will be replaced with "aaa authorization serial-console". 
Seems like it was the assumption that "aaa authentication console" command will be also replaced in the same way, but this is not what happened.
Under EOS 4.24:
#aaa authentication logi ?
  console  Console authentication list
  default  The default authentication list

Currently, aaa authentication module supports only "default" and "serial-console" options: https://avd.sh/en/latest/roles/eos_cli_config_gen/index.html#aaa-authentication

Serial-console is accepted because it is a hidden command, but it will not be applied to console. For this reason, "serial_console" should be replaced with "console" along with the EOS output command.

This is a breaking change, but should be considered since current generated command is not generating expected behaviour.

## Component Name
eos_cli_config_gen

## Steps to reproduce

```yaml
aaa_authentication:
  login:
    serial_console: local group radius_servers
```

### Expected results

```shell
aaa authentication login console local group radius_servers
```

### Actual results

```shell
aaa authentication login serial-console local group radius_servers
```

### PR DESCRIPTION
**Pull request by xaviramon**
## Change Summary

AAA authentication was rendered to a CLI syntax that was not being applied to serial interface. This fix changes the structure to be self-explanatory and also generates the proper CLI configuration.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #1099 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
```aaa_authentication:
  login:
    default: group TACACS local
    serial_console: local
```
Is changed to:
```aaa_authentication:
  login:
    default: group TACACS local
    console: local
```
## How to test
generate the structure and apply to device/host vars

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Review by titom73 — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]
LGTM - verified on EOS that old CLI is accepted but is not working as intended. New CLI is working as it should. Thank you for finding this and providing a fix.

**Comment by kmueller68** [comment]
Hi.
Would you please also forward this information to the Arista department for the EOS manuals?
EOS 4.26.0F manual states the following:
```
8.1.1.7.13 aaa authorization serial-console
The aaa authorization serial-console command configures the switch to authorize

switch(config)#aaa authorization serial-console
switch(config)#
```

**Comment by xaviramon** [comment]
> Hi.
> Would you please also forward this information to the Arista department for the EOS manuals?
> EOS 4.26.0F manual states the following:
> 
> ```
> 8.1.1.7.13 aaa authorization serial-console
> The aaa authorization serial-console command configures the switch to authorize
> 
> switch(config)#aaa authorization serial-console
> switch(config)#
> ```

Hi Karl,
Thank you for your comment.

This is a confusing topic, but the conclusion is the documentation you reference is right. The key is two different commands doing a very similar function (aaa authorization and aaa authentication) are diverging in terms of configuration. So let me explain:
On EOS versions up to 4.22, both authorization and authentication commands are configured on console with this keyword:
```yaml
aaa authorization console
aaa authentication console
```
This behaviour is broken because of Field Notice 39 ( https://www.arista.com/en/support/advisories-notices/fieldnotices/7097-field-notice-39 )[registration needed]. In this FN, following change is announced:
```yaml
aaa authorization console
```
to
```yaml
aaa authorization serial-console
```
Note that authentication command is NOT modified.

In conclusion:
Starting EOS 4.23, valid configuration commands for authorization and authentication on console port are:
```yaml
aaa authorization serial-console
aaa authentication console
```

This PR solves the issue that authentication command on console port was rendered wrong, but still, accepted, even when behaviour of device was not the expected one:
```yaml
aaa authentication serial-console
```
In my opinion, and some other opinions from my colleagues, both commands should use same syntax. And this is the feedback I am planing to send to software team.

Please, let me know your thoughts on this.

Thank you

**Comment by kmueller68** [comment]
@xaviramon Hi Xavier,
I think I got you. Having the same syntax for authentication and authorization makes definitely sense.

With the current implementation I will end up on this:
```
aaa authentication login default group AAA_Auth local
aaa authentication login console group AAA_Auth local
!
aaa authorization exec default group AAA_Auth local
aaa authorization serial-console
```
